File: API/src/main/java/com/concierge/apiconcierge/models/part/group/GroupPart.java
package com.concierge.apiconcierge.models.part.group;

import com.concierge.apiconcierge.models.enums.StatusEnableDisable;
import com.concierge.apiconcierge.models.part.enums.TypeGroupPart;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
@SecondaryTable(name = "tb_company", pkJoinColumns = @PrimaryKeyJoinColumn(name = "id"))
@SecondaryTable(name = "tb_resale", pkJoinColumns = @PrimaryKeyJoinColumn(name = "id"))
@Entity
@Table(name = "tb_part_group")
public class GroupPart {

    @JoinColumn(table = "tb_company", referencedColumnName = "id")
    @Column(name = "company_id")
    private Integer companyId;

    @JoinColumn(table = "tb_resale", referencedColumnName = "id")
    @Column(name = "resale_id")
    private Integer resaleId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private StatusEnableDisable status;

    private TypeGroupPart type;

    private String description;

    @JoinColumn(table = "tb_brand", referencedColumnName = "id")
    @Column(name = "brand_id")
    private Integer brandId;
}

File: Front-End/src/app/models/parts/group.part.ts
import { StatusEnabDisabEnum } from "../status-enab-disab-enum";
import { GroupPartTypeEnum } from "./group.part.type.enum";

export class GroupPart {
    companyId: number | null = null;
    resaleId: number | null = null;
    id: number | null = null;
    status: StatusEnabDisabEnum = StatusEnabDisabEnum.DISABLED;
    type: GroupPartTypeEnum = GroupPartTypeEnum.OUTROS;
    description: string = "";
    brandId: number | null = null;
}